### ISSUE TITLE
TypeScript hot reload doesn't work

### ISSUE THREAD

**Issue by iljaiwjew** [Azure/azure-functions-core-tools#3239]
I init the project with VS Code extension and hot reload doesn't work if run **func start** or **npm run start** which is actually the same but it works if i run **npm run start** and **tsc -w** in the second terminal window

**Comment by flo-sch** [comment]
I do not believe azure-function-core-tools comes with a built-in hot reload feature at the moment (see #1239)

But if you use node.js, you can leverage a tool like [concurrently](https://www.npmjs.com/package/concurrently) to run the commands you mention in parallel, using a single terminal window.

NOTE: This pattern has worked fine for me those past years, however I started noticing issues with the latest release `4.0.5085` of AZCT (released 2 days ago), so I am sticking to `4.0.5030` for now.

Here is an example of a package.json containing a `watch` scripts running it:

```json
{
  "devDependencies": {
    "azure-function-core-tools": "^4.0 <=4.0.5030",
    "concurrently": "^7.6.0"
  },
  "scripts": {
    "watch": "concurrently \"tsc -w\" \"func start --typescript\""
  }
}
```

I hope issues with 4.0.5085 will be resolved (or the feature eventually supported natively by AZCT, but I guess it is not easy to natively support that for every language 🤷)

**Comment by lfantone** [comment]
I was having the same issue, "_hot reload_" stop working after upgrading to latest `azure-function-core-tools`. Not sure what was the changes behind it.
Meanwhile, I fixed the dependency version to **4.0.5030**.

**Comment by ejizba** [comment]
@flo-sch @lfantone would you mind trying v4.0.5095? It was just released this morning and I believe will fix your issue

**Comment by flo-sch** [comment]
First thing tomorrow morning, sure, I will let you know 😀

**Comment by philhagger** [comment]
So I'm running v4.0.5095 trying to run ES6 module style *.mjs files and any changes I make to files are still not triggering hot reloading. I am not using TypeScript.

``

**Comment by philhagger** [comment]
To add to that, I have tried reverting the *.mjs ES6 code and I am still not getting any Hot reloading, this has always worked before I tried the new modelv4 preview...

**Comment by ejizba** [comment]
@philhagger hot reload does not work in the new model yet. We'll use the issue you created to track this for GA: https://github.com/Azure/azure-functions-nodejs-worker/issues/674

**Comment by ejizba** [comment]
@gabrieljoelc  Yes this should've been fixed already in v4.0.5198, even for the new model. I forgot to close this issue at the time I made that comment.

Since this is a pretty old issue and was originally for the old model, would you mind creating a new issue with detailed repro steps. This would be my preferred repository for the issue: https://github.com/Azure/azure-functions-nodejs-library/issues
